### PR TITLE
[release-4.14] OCPEDGE-1089: chore: use rhel9 for the CI build root

### PR DESCRIPTION
**Pull request by jeff-roche**
Cherry-pick https://github.com/openshift-kni/lifecycle-agent/pull/590 to release-4.16

### CODE REVIEW

File: .ci-operator.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-8-release-golang-1.21-openshift-4.16
+  tag: rhel-9-release-golang-1.21-openshift-4.16
```
Comment (donpenney) [inline .ci-operator.yaml:4]: If this is backported, this should probably be a 4.14 image ref?

Comment (donpenney) [inline .ci-operator.yaml:4]:
```suggestion
  tag: rhel-9-release-golang-1.21-openshift-4.14
```

Comment (jeff-roche) [reply .ci-operator.yaml]: This is the same issue as 4.15, there is no go lang 1.21 image prior to 4.16

Comment (donpenney) [reply .ci-operator.yaml]: ok

File: .ci-operator.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-8-release-golang-1.21-openshift-4.16
+  tag: rhel-9-release-golang-1.21-openshift-4.14
```
Comment (donpenney) [inline .ci-operator.yaml:4]: Undoing my previous edit, per conversation:
```suggestion
  tag: rhel-9-release-golang-1.21-openshift-4.16
```

**Review by donpenney — APPROVED** [review]

**Pull request merged**